lib/operations/find_and_modify.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const OperationBase = require('./operation').OperationBase;
+const applyRetryableWrites = require('../utils').applyRetryableWrites;
+const applyWriteConcern = require('../utils').applyWriteConcern;
+const decorateWithCollation = require('../utils').decorateWithCollation;
+const executeCommand = require('./db_ops').executeCommand;
+const formattedOrderClause = require('../utils').formattedOrderClause;
+const handleCallback = require('../utils').handleCallback;
+const ReadPreference = require('mongodb-core').ReadPreference;
+
+class FindOneAndDeleteOperation extends OperationBase {
+  constructor(collection, query, sort, doc, options) {
+    super(options);
+
+    this.collection = collection;
+    this.query = query;
+    this.sort = sort;
+    this.doc = doc;
+  }
+
+  execute(callback) {
+    const coll = this.collection;
+    const query = this.query;
+    let sort = this.sort;
+    const doc = this.doc;
+    const options = this.options;
+
+    // Create findAndModify command object
+    const queryObject = {
+      findAndModify: coll.s.name,
+      query: query
+    };
+
+    sort = formattedOrderClause(sort);
+    if (sort) {
+      queryObject.sort = sort;
+    }
+
+    queryObject.new = options.new ? true : false;
+    queryObject.remove = options.remove ? true : false;
+    queryObject.upsert = options.upsert ? true : false;
+
+    const projection = options.projection || options.fields;
+
+    if (projection) {
+      queryObject.fields = projection;
+    }
+
+    if (options.arrayFilters) {
+      queryObject.arrayFilters = options.arrayFilters;
+      delete options.arrayFilters;
```

Comment:
does this line mutate the user-passed options object?

Reply:
+1. I think we should remove the `delete`